server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodesInfoResponse.java

```diff
@@ -66,6 +66,10 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.field("version", nodeInfo.getVersion());
             builder.field("transport_version", nodeInfo.getTransportVersion().id());
             builder.field("index_version", nodeInfo.getIndexVersion().id());
+            builder.startObject("component_versions");
+            for (var cv : nodeInfo.getComponentVersions().entrySet()) {
+                builder.field(cv.getKey(), cv.getValue());
```

Comment:
Should we entirely separate the data from structure here? So have `[{ component: "MlConfigVersion", version: 12345 }, ...]` instead?

Reply:
That's an interesting question. I think the only advantage would be consistency, right?
Like you said, we have individual `transport_version: 123` and `index_version: 123`, so it would be nice to read `ml_config_version: 1234` etc. On the other hand, they are already `component_versions`, so having a 
```
transport_version: 123,
component_versions: [ 
   { component: "MlConfigVersion", version: 12345 }
]
``` 
object might have sense.
All in all, I think the latter might be clearer, but I don't feel about it strongly.

Reply:
I would stick with the component_versions as an object. Can we also at least document, if not enforce, snake_case on the id returned?

Reply:
Added javadoc & an assertion check